### DISCUSSION TITLE
Trouble using multiple gateways

### DISCUSSION THREAD

**Discussion by MK-Mees** [duncanmcclean/simple-commerce#1075] [Help]
Hi!

I am currently developing a checkout page where the customer fills in their information and then selects a paymentmethod:
`$checkout = Statamic::tag('sc:checkout')->params(['redirect' => '/afronden'])->fetch();`

There are two payment methods. Mollie and Cash on Pickup which are configured in the simple-commerce.php:
```
'gateways' => [
        \App\Gateways\CashOnPickup::class => [],
        \DuncanMcClean\SimpleCommerce\Gateways\Builtin\MollieGateway::class => [
            'key' => env('MOLLIE_KEY'),
            'profile' => env('MOLLIE_PROFILE'),
        ],
    ],
```

This is how i fetch the gateways:
`$gateways = Statamic::tag('sc:gateways')->fetch();`
Which are in turn loaded into the form like this:
```
<select id="gateway" name="gateway" class="w-full" required>
      @foreach($gateways as $gateway)
          <option value="{{ $gateway['class'] }}">{{ $gateway['display'] }}</option>
      @endforeach
  </select>
```

Whenever i select either gateway it gives me an error:
**The provided payment gateway is not valid.**

I did try testing the Mollie payment gateway using:
```
$form = Statamic::tag('sc:checkout:mollie')->params([
      'redirect' => '/afronden',
      'error_redirect' => "/foutje"
])->fetch();
```
Which seems to work as intended.

Do you have any idea what i'm doing wrong?
Thanks in advance!

**Comment by duncanmcclean** [comment]
Can you try using `{{ $gateway['handle'] }}` as the option values instead?

**Reply by MK-Mees** [reply]
That worked for Cash on Pickup! Thanks @duncanmcclean.

But when selecting Mollie it now says:
**This gateway does not support the [checkout] method.**

Is it expected to catch a Mollie gateway and re-route it to it's own file which contains a Mollie tag?
```
$form = Statamic::tag('sc:checkout:mollie')->params([
      'redirect' => '/afronden',
      'error_redirect' => "/foutje"
])->fetch();
```

**Reply by duncanmcclean** [reply]
> Is it expected to catch a Mollie gateway and re-route it to it's own file which contains a Mollie tag?

Yes, that's the expected way to do it. It's also the way we do it in the starter kit.
